### PR TITLE
Update structure

### PR DESCRIPTION
**Pull request by codingjeremy**

### CODE REVIEW

**Comment by codingjeremy** [comment]
Most changes are in the MainActivity and PermissionRationalActivity files. Most of the others were just moving this to a complete module (vs app) so I can create a start/init module for starting the codelab.

Either way, codelab is working now on 29.

File: complete/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+/build
```
Comment (thagikura) [inline complete/.gitignore]: Is this file needed? Because the same entry is in the top-level .gitignore.

File: complete/src/main/java/com/google/example/android/basicactivityrecognitiontransitionsample/MainActivity.java [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+        // TODO: Initialize PendingIntent that will be triggered when a activity transition occurs.
+        Intent intent = new Intent(TRANSITIONS_RECEIVER_ACTION);
+        mActivityTransitionsPendingIntent =
+                PendingIntent.getBroadcast(MainActivity.this, 0, intent, 0);
+
+        // TODO: Create and register a BroadcastReceiver to listen for activity transitions.
+        // The receiver listens for the PendingIntent above that is triggered by the system when an
+        // activity transition occurs.
+        mTransitionsReceiver = new TransitionsReceiver();
+        registerReceiver(mTransitionsReceiver, new IntentFilter(TRANSITIONS_RECEIVER_ACTION));
```
Comment (thagikura) [inline complete/src/main/java/com/google/example/android/basicactivityrecognitiontransitionsample/MainActivity.java]: Can this be via LocalBroadcastManager to reduce the risk of leaking the BroadcastReceiver?
I'm not familiar with the ActivityRecognition, so I'm just checking in case.

File: complete/build.gradle [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+dependencies {
+    implementation fileTree(dir: 'libs', include: ['*.jar'])
+
+    implementation 'androidx.annotation:annotation:1.1.0'
+
+    implementation 'androidx.appcompat:appcompat:1.1.0'
+    implementation 'androidx.constraintlayout:constraintlayout:1.1.3'
+    implementation 'com.google.android.material:material:1.0.0'
+
+    // TODO: Add play services library required for activity recognition.
```
Comment (thagikura) [inline complete/build.gradle]: If this module is a complete example, are these TODOs needed?

Comment (codingjeremy) [reply complete/build.gradle]: TODOs are needed when I copy this complete module to a "start" module. I then remove the code below it. I also want devs to easily find in the complete module where the code is in case they make a mistake in the "start" module.

File: complete/src/main/java/com/google/example/android/basicactivityrecognitiontransitionsample/MainActivity.java (hunk 2 above)
Comment (codingjeremy) [reply complete/src/main/java/com/google/example/android/basicactivityrecognitiontransitionsample/MainActivity.java]: Good point, I was using their original code for that part. I will rework it.

File: complete/.gitignore (hunk 1 above)
Comment (codingjeremy) [reply complete/.gitignore]: Removed.

File: complete/src/main/java/com/google/example/android/basicactivityrecognitiontransitionsample/MainActivity.java (hunk 2 above)
Comment (codingjeremy) [reply complete/src/main/java/com/google/example/android/basicactivityrecognitiontransitionsample/MainActivity.java]: Fixed.
